### ISSUE TITLE
scipy.signal.residue reports an incorrect result in the case of repeating roots.

### ISSUE THREAD

**Issue by ivorjawa** [scipy/scipy#6275]
A partial fraction expansion using residue() fails on polynomials that have repeated complex roots.
I've taken test cases from  Chapter 12 of Nilsson's Electric Circuits, 9th Edition
The particular function is as follows:

```
# Repeated Complex Roots, case 4, P 451
Frc = 768 / (s**2 + 6*s +25)**2
Frc

768/(s**2 + 6*s + 25)**2
```

Expanded, this yields the following numerator and denominator:

```
nump: [ 768.]
denp: [   1.   12.   86.  300.  625.]
```

scipy.signal.residue fails as follows:

```
parfrac(Frc) # incorrect.  Roots are correct, constants are wrong. p451
# should be -12, -3j, -12, 3j
# note warnings

/Users/kujawa/Projects/CatHouseStudio/CV2/environment/opencv_env/lib/python2.7/site-packages/scipy/signal/signaltools.py:1514: RuntimeWarning: divide by zero encountered in cfloat_scalars
  / factorial(sig - m))
/Users/kujawa/Projects/CatHouseStudio/CV2/environment/opencv_env/lib/python2.7/site-packages/scipy/signal/signaltools.py:1514: RuntimeWarning: invalid value encountered in cfloat_scalars
  / factorial(sig - m))
/Users/kujawa/Projects/CatHouseStudio/CV2/environment/opencv_env/lib/python2.7/site-packages/scipy/signal/signaltools.py:1514: RuntimeWarning: invalid value encountered in true_divide
  / factorial(sig - m))

(array([ nan+nanj,  nan+nanj,  nan+nanj,  nan+nanj], dtype=complex64),
 array([-3.+4.j, -3.-4.j, -3.+4.j, -3.-4.j], dtype=complex64),
 array([ 0.], dtype=float32))
```

I've attached a jupyter notebook demonstrating all four test cases.

The code:

``` python
def polyarray(poly):
    # translates a polynomial into an array of coefficients
    ap = sym.apart(poly)
    try:
        retval =  sym.poly(ap).all_coeffs()
    except sym.GeneratorsNeeded, e:
        retval = [float(ap)]
    return np.array(retval, dtype=np.float32)

def translate(poly):
    # translates an S-domain transfer function into a TransferFunction
    num, den = sym.fraction(poly)
    nump = polyarray(num)
    print "nump: %s" % str(nump)
    denp = polyarray(den)
    print "denp: %s" % str(denp)
    return signal.TransferFunction(nump, denp)

def parfrac(poly):
    # computes a partial fraction expansion of an S-domain transfer function.
    tf = translate(poly)
    return signal.residue(tf.num, tf.den)
```

[bug_report.zip](https://github.com/scipy/scipy/files/317177/bug_report.zip)

**Comment by roryyorke** [comment]
I think this has been fixed, probably by #10997; see below for a test with Scipy 1.6.0, installed from PyPI on Ubuntu 20.04. 

```
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)
1.6.0 1.19.5 sys.version_info(major=3, minor=8, micro=5, releaselevel='final', serial=0)
>>> numpy.set_printoptions(precision=3, suppress=True)
>>> from scipy.signal import residue
>>> for res in residue(768, [1, 12, 86, 300, 625]):
...     print(res)
... 
[  0.-3.j -12.-0.j   0.+3.j -12.+0.j]
[-3.+4.j -3.+4.j -3.-4.j -3.-4.j]
[]
```

**Comment by ilayn** [comment]
This is still a no repro on main. Hence closing.
